File: Modul-6/Soal-4/PRAK604-2310817110013-Randy_Febrian.c
#include <stdio.h>
#include <string.h>

int main(){
    char pesanAsli [50];
    char pesanMasuk [50];
    int bintang = 0;
    int pagar = 0;

    fgets(pesanAsli, sizeof(pesanAsli), stdin);
    fgets(pesanMasuk, sizeof(pesanMasuk), stdin);

    if(strlen(pesanAsli) != strlen(pesanMasuk)){
        printf("Panjang kalimat berbeda, pesan palsu");
        return 1;
    }

    for(int i = 0; i < strlen(pesanAsli) - 1; i++){
        if(pesanAsli[i] == pesanMasuk[i]){
            if(pesanAsli[i] == ' ' && pesanMasuk[i] == ' '){
                printf(" ");
            }
            else if(pesanAsli[i] != ' ' && pesanMasuk[i] != ' '){
                bintang++;
                printf("*");
            }
        }
        else if(pesanAsli[i] != pesanMasuk[i]){
           pagar++;
           printf("#");
        }

    }
    printf("\n");
    printf("* = %d\n", bintang);
    printf("# = %d\n", pagar);

    if(pagar > bintang){
        printf("pesan palsu\n");
    }
    else{
        printf("pesan asli\n");
    }
    return 0;
}

File: Modul-6/Soal-4/PRAK604-2310817110013-Randy_Febrian.py
pesanAsli = input()
pesanMasuk = input()
bintang = 0
pagar = 0

if(len(pesanAsli) != len(pesanMasuk)):
    print("Panjang kalimat berbeda, pesan palsu")
    exit()

for i in range (len(pesanAsli)):
    if pesanAsli[i] == pesanMasuk[i] and pesanAsli[i] != ' ':
        if pesanAsli[i] == ' ' and pesanMasuk[i] == ' ':
            print(" ", end='')
        elif pesanAsli[i] != ' ' and pesanMasuk[i] != ' ':
            bintang += 1
            print("*", end='')
    elif pesanAsli[i] != pesanMasuk[i]:
        pagar += 1
        print("#", end='')
print()
print("* = %d" % bintang)
print("# = %d" % pagar)

if pagar > bintang:
    print("Pesan Palsu")
else:
    print("Pesan Asli")
